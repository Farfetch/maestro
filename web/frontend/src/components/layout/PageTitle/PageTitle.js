import { Col, Row, Typography } from "antd";
import PropTypes from "prop-types";

const { Title } = Typography;

const PageTitleRow = ({ title, button = null }) => (
  <Row style={{ marginBottom: "1rem" }}>
    <Col flex="auto">
      <Title style={{ marginBottom: "0" }}>{title}</Title>
    </Col>
    {button ? (
      <Col
        flex="100px"
        style={{
          alignItems: "center",
          display: "flex",
          justifyContent: "right"
        }}
      >
        {button}
      </Col>
    ) : null}
  </Row>
);

PageTitleRow.propTypes = {
  title: PropTypes.string.isRequired,
  button: PropTypes.any
};

export default PageTitleRow;
