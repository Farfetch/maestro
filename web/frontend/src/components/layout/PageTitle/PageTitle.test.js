import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import PageTitle from "./PageTitle";

describe("components/PageTitle", () => {
  [
    {
      title: "Test title"
    },
    {
      title: "Test title",
      button: {
        link: "/button-test-link",
        title: "Button title"
      }
    }
  ].map((props) =>
    test(`should render PageTitle component`, async () => {
      const rendered = render(
        <MemoryRouter initialEntries={["/tests"]}>
          <PageTitle {...props} />
        </MemoryRouter>
      );

      const badgeComponent = rendered.container;
      expect(badgeComponent.outerHTML).toMatchSnapshot();
    })
  );
});
