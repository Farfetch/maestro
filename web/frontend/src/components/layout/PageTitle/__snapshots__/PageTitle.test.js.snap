// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PageTitle should render PageTitle component 1`] = `"<div><div class=\\"ant-row css-dev-only-do-not-override-nnuwmp\\" style=\\"margin-bottom: 1rem;\\"><div style=\\"flex-basis: auto;\\" class=\\"ant-col css-dev-only-do-not-override-nnuwmp\\"><h1 class=\\"ant-typography css-dev-only-do-not-override-nnuwmp\\" style=\\"margin-bottom: 0px;\\">Test title</h1></div></div></div>"`;

exports[`components/PageTitle should render PageTitle component 2`] = `"<div><div class=\\"ant-row css-dev-only-do-not-override-nnuwmp\\" style=\\"margin-bottom: 1rem;\\"><div style=\\"flex-basis: auto;\\" class=\\"ant-col css-dev-only-do-not-override-nnuwmp\\"><h1 class=\\"ant-typography css-dev-only-do-not-override-nnuwmp\\" style=\\"margin-bottom: 0px;\\">Test title</h1></div><div style=\\"flex: 0 0 100px; align-items: center; display: flex; justify-content: right;\\" class=\\"ant-col css-dev-only-do-not-override-nnuwmp\\"><a href=\\"/button-test-link\\"><button type=\\"button\\" class=\\"ant-btn css-dev-only-do-not-override-nnuwmp ant-btn-primary ant-btn-lg\\"><span>Button title</span></button></a></div></div></div>"`;
